docker-compose.yml

```diff
@@ -0,0 +1,10 @@
+version: '3'
+services:
+  api:
+    build: ./api
+    ports: ["3001:3001"]
+    volumes: ['./api:/app']
+  client:
+    build: ./client
+    ports: ["3000:3000"]
+    volumes: ['./client:/app']
```

Comment:
@nazar with two different docker containers, the react client is now not able to proxy requests to the server api using the value localhost:3000 as defined in the `package.json`

If you change this to `http://api:3001/` then it works as services [are able to reference](https://docs.docker.com/compose/compose-file/#links) other services. 

I tried setting [npm environment variables](https://docs.npmjs.com/misc/config#environment-variables) but I couldn't get this to work. It would be nice to find a solution so that this continues to work with either docker or not, and without needing to remember to set/change a specific value. any ideas?

Reply:
I'll need to have a closer look at Docker as I'm seeing other issues such as `docker-compose up` not consistently starting the client in a correct state and IO performance issues on the shared folder, particularly in the client due to the large node_modules folder.

Reply:
>If you change this to http://api:3001/

So this works:

```
    links:
       - api:localhost
```